### DISCUSSION TITLE
Add label to shared-links

### DISCUSSION THREAD

**Discussion by andyhoyland** [plausible/analytics#795] [Anything Else]
Could we have a name label for the shared links?

I'm in the position where I create different shared links for different people (or groups of people) to see a site's stats.

I have one link to share with the internal team involved in the site, a link to share with the person who comissioned the site, and now a link to share with an external agency who are creating social ads and things to bring people to the site. The external group only need access to the stats for a limited time period (the duration of their ads campaign). Sharing different links with different groups means I can remove one link without having to send out a new link to the remaining people.

Something like the below.

![Screenshot 2021-03-05 at 08 59 50](https://user-images.githubusercontent.com/2813738/110092248-4e296e00-7d91-11eb-8c78-2423800e320c.png)

(First discussion starter - sorry if I got this all wrong!)

**Comment by ukutaht** [comment]
Yes! I actually though about adding it in the last round of updates to shared links but wasn't sure if it would be useful or not. Also, I wasn't sure how to handle existing links, I guess we could auto-generate something but it would not be very descriptive.

I would gladly merge a PR for this :)

Design wise, could also be nice if the action buttons were on the descriptive label instead of the raw link. Come to think of it, once we have labels is it even necessary to show the raw link? Something I'd need to test out both ways to see which one feels better. Showing both is very safe though.

**Comment by ukutaht** [comment]
This is now available on our cloud version. There are labels (which are auto-generated from the link slug itself) along with an icon to show whether the shared link is password-protected or not. There's also a new icon to edit the shared link so you can rename your existing links to something meaningful. Let me know what you think!

![Screenshot from 2021-04-06 15-26-09@2x](https://user-images.githubusercontent.com/3731516/113710561-8a335400-96ec-11eb-8b9d-426db6ff3e4e.png)

**Reply by andyhoyland** [reply]
Thanks for that!

That's great... wish I'd been able to contribute but really pleased you were able to make it happen.
